vibin/mediaservers/cxnv2.py

```diff
@@ -0,0 +1,589 @@
+import base64
+import time
+from collections import deque, Counter
+from pathlib import Path
+from typing import Iterable
+from xml.etree.ElementTree import Element
+
+import upnpclient
+from urllib.request import urlopen
+from urllib.parse import urlparse
+from xml.etree import ElementTree
+from functools import cache
+
+from vibin import VibinNotFoundError
+from vibin.logger import logger
+from vibin.mediaservers import MediaServer
+from vibin.models import MediaServerState, Album, Artist, Track, UPnPServiceSubscriptions, MediaBrowseSingleLevel, \
+    MediaFolder
+from vibin.types import UpdateMessageHandler, MediaId, UPnPProperties, MediaType
+
+# -----------------------------------------------------------------------------
+# Implementation of MediaServer, using USB drives attached to a CXNv2.
```

Comment:
These implementation notes are super helpful (and interesting), thanks!

Reply:
Thanks :)